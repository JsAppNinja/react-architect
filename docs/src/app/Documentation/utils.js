export const splitOnCaps = s => s.split(/(?=[A-Z1-9])/);

function getString(s) {
  return !(s instanceof Array) ? splitOnCaps(s) : s;
}

export const toDashedName = (s) => getString(s).map(s => s.toLowerCase()).join('-');
export const toTitle = (s) => getString(s).join(' ');
