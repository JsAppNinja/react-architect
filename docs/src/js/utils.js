import React from 'react';

import { FontIcon, Avatar } from 'react-md';

export const githubHref = 'https://github.com/mlaursen/react-md';

export const mainLinks = [{
  link: '',
  primaryText: 'Home',
  leftIcon: <FontIcon>home</FontIcon>,
}, {
  link: 'sassdoc',
  primaryText: 'SASS Doc',
  leftAvatar: <Avatar src="/imgs/sass-icon.png" alt="SASS Icon" />,
}];
