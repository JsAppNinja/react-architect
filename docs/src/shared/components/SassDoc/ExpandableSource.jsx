import React, { PureComponent, PropTypes } from 'react';
import { findDOMNode } from 'react-dom';
import AccessibleFakeButton from 'react-md/lib/Helpers/AccessibleFakeButton';
import { Motion, spring } from 'react-motion';

import lineType from './lineType';
import ScssMarkdown from './ScssMarkdown';

const INITIAL_HEIGHT = 83;

function isMultiline(source) {
  return source && !!source.match(/\r?\n/);
}

function extractSource(source, { start: sourceStart, end: sourceEnd }, { end: commentEnd } = {}) {
  const start = commentEnd ? commentEnd + 1 : sourceStart;

  const split = source.split(/\r?\n/);

  if (start === sourceEnd) {
    return split[start - 1];
  }

  return split.slice(start - 1, sourceEnd).join('\n');
}

function trim(string) {
  return string.replace(/\r?\n/g, '').replace(/ {2}/, '').replace(/ {2}/g, ' ');
}

const MAP_SPLIT = ': (';
const MIXIN_SPLIT = '{';

function hideSource(source) {
  let s;
  let delimiter = '}';
  if (!source.match(/@mixin/)) {
    delimiter = ')';
    s = source.substring(0, source.indexOf(MAP_SPLIT) + MAP_SPLIT.length);
  } else {
    s = source.substring(0, source.indexOf(MIXIN_SPLIT) + MIXIN_SPLIT.length);
  }

  return `${trim(s)} \u2026 ${delimiter}`;
}

function getSelectedText() {
  if (typeof window.getSelection === 'function') {
    return window.getSelection().toString();
  } else if (typeof document.selection !== 'undefined' && document.selection.type === 'Text') {
    return document.selection.createRange().text;
  }

  return '';
}

export default class ExpandableSource extends PureComponent {
  static propTypes = {
    commentRange: lineType,
    line: lineType,
    rawFile: PropTypes.string.isRequired,
  };

  constructor(props) {
    super(props);

    const source = extractSource(props.rawFile, props.line, props.commentRange);
    const oneLineSource = hideSource(source);
    this.state = {
      source,
      height: INITIAL_HEIGHT,
      decrementing: false,
      oneLineSource,
      visibleSource: isMultiline(source) ? oneLineSource : source,
    };

    this._setHeight = this._setHeight.bind(this);
    this._setButton = this._setButton.bind(this);
    this._toggleSource = this._toggleSource.bind(this);
    this._maybeToggleSource = this._maybeToggleSource.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.rawFile !== nextProps.rawFile) {
      const source = extractSource(nextProps.rawFile, nextProps.line, nextProps.commentRange);
      const oneLineSource = hideSource(source);
      this.setState({
        source,
        oneLineSource,
        visibleSource: isMultiline(source) ? oneLineSource : source,
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    const { height, visibleSource } = this.state;
    if (height === INITIAL_HEIGHT && visibleSource !== prevState.visibleSource) {
      this._setHeight();
    }
  }

  _setButton(button) {
    this._button = findDOMNode(button);
  }

  _setHeight() {
    if (!this._button) {
      return;
    }

    this.setState({ height: this._button.childNodes[0].offsetHeight + 32 });
  }

  _toggleSource() {
    if (this._button) {
      const text = getSelectedText();
      if (text && this._button.textContent.indexOf(text) !== -1) {
        return;
      }
    }

    if (this.state.height > INITIAL_HEIGHT) {
      this.setState({ height: INITIAL_HEIGHT, decrementing: true });
    } else {
      this.setState({ visibleSource: this.state.source });
    }
  }

  _maybeToggleSource() {
    if (this.state.height === INITIAL_HEIGHT) {
      this.setState({ visibleSource: this.state.oneLineSource, decrementing: false });
    }
  }

  render() {
    const { visibleSource, height, oneLineSource, decrementing, source } = this.state;
    if (!isMultiline(source)) {
      return (
        <AccessibleFakeButton
          component={ScssMarkdown}
          disabled
          className="md-paper md-paper--1 md-card-text expandable-source"
          markdown={source}
        />
      );
    }

    return (
      <Motion
        style={{ height: spring(height) }}
        defaultStyle={{ height: INITIAL_HEIGHT }}
        onRest={this._maybeToggleSource}
      >
        {style => {
          let markdown = visibleSource;
          if (style.height < 100 && decrementing) {
            markdown = oneLineSource;
          }

          return (
            <div>
              <AccessibleFakeButton
                ref={this._setButton}
                style={style}
                onClick={this._toggleSource}
                component={ScssMarkdown}
                className="md-paper md-paper--1 md-card-text expandable-source"
                markdown={markdown}
              />
              <h5 className="md-cell--right md-color--secondary-text md-text-right">
                {`Click to ${decrementing || visibleSource === oneLineSource ? 'expand' : 'collapse'}`}
              </h5>
            </div>
          );
        }}
      </Motion>
    );
  }
}
