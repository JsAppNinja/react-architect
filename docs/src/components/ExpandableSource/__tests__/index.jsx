/* eslint-disable max-len */
/* eslint-env jest */
import React from 'react';
import { createRouterSnapshot } from 'utils/testing';

import ExpandableSource from '../';

const ONE_LINE_CODE = '$md-primary-color: $md-red-500;';
const CODE = `@mixin react-md-buttons(
  $primary-color: $md-primary-color,
  $secondary-color: $md-secondary-color,
  $light-theme: $md-light-theme,
  $include-media: $md-media-included,
  $include-flat: $md-btn-include-flat,
  $include-raised: $md-btn-include-raised,
  $include-icon: $md-btn-include-icon,
  $include-floating: $md-btn-include-floating,
  $include-dense-icons: $md-font-icon-include-dense
) {
  // scss-lint:disable QualifyingElement
  a.md-btn {
    text-decoration: none;
  }

  .md-btn {
    background: transparent;
    border: 0;
    position: relative;
    transition-duration: $md-transition-time;
    transition-property: background, color;

    // Prevents click event from firing when clicking a child
    &[disabled] * {
      pointer-events: none;
    }

    &:focus {
      outline-style: none;
    }

    .md-icon {
      color: inherit;
    }

    .md-icon-separator {
      height: 100%;
    }

    .md-icon-text {
      @extend %md-font-medium;
    }
  }

  .md-btn--hover {
    background: if($light-theme, $md-btn-light-theme-hover-color, $md-btn-dark-theme-hover-color);
  }

  .md-btn--color-primary-active {
    background: rgba($primary-color, .12);
  }

  .md-btn--color-secondary-active {
    background: rgba($secondary-color, .12);
  }

  @if $include-flat or $include-raised {
    .md-btn--text {
      border-radius: $md-btn-text-border-radius;
      font-size: $md-btn-text-font-size;
      font-weight: $md-font-medium;
      height: $md-btn-text-height;
      min-width: $md-btn-min-width;
      padding: $md-btn-tb-padding $md-btn-lr-padding;
      text-transform: uppercase;
    }
  }

  @if $include-raised or $include-floating {
    .md-btn--raised {
      @include md-box-shadow(1);

      transition: background $md-transition-time, box-shadow $md-transition-time * 2, color $md-transition-time;
    }

    .md-btn--raised-disabled {
      background: get-color('divider', $light-theme);
    }

    .md-btn--raised-pressed {
      @include md-box-shadow(2);
    }
  }

  @if $include-icon or $include-floating {
    .md-btn--icon {
      border-radius: $md-btn-icon-border-radius;
      color: get-color('icon', $light-theme);
      height: $md-btn-icon-size;
      padding: $md-btn-icon-padding;
      width: $md-btn-icon-size;
    }
  }

  @if $include-floating {
    .md-btn--floating {
      @include md-box-shadow(2);

      height: $md-btn-floating-size;
      padding: ($md-btn-floating-size - $md-font-icon-size) / 2;
      transition-property: background, box-shadow, color;
      width: $md-btn-floating-size;
    }

    .md-btn--floating-pressed {
      @include md-box-shadow(4);
    }

    .md-btn--floating-mini {
      height: $md-btn-floating-mini-size;
      padding: ($md-btn-floating-mini-size - $md-font-icon-size) / 2;
      width: $md-btn-floating-mini-size;
    }

    .md-btn--fixed {
      position: fixed;
      z-index: $md-btn-fixed-z-index;
    }

    @if $md-btn-floating-margin != null {
      @include react-md-button-fixed-positions($md-btn-floating-margin);
    }
  }

  @if $include-media {
    @include react-md-buttons-media($include-flat, $include-raised, $include-icon, $include-floating, $include-dense-icons);
  }
}`;

const CODE_LINE = '@mixin react-md-buttons($primary-color: $md-primary-color, $secondary-color: $md-secondary-color, $light-theme: $md-light-theme, $include-media: $md-media-included, $include-flat: $md-btn-include-flat, $include-raised: $md-btn-include-raised, $include-icon: $md-btn-include-icon, $include-floating: $md-btn-include-floating, $include-dense-icons: $md-font-icon-include-dense) { ... }';

describe('ExpandableSource', () => {
  it('should render correctly while collapsed', () => {
    const tree1 = createRouterSnapshot(<ExpandableSource code={ONE_LINE_CODE} oneLineCode={ONE_LINE_CODE} />);
    expect(tree1).toMatchSnapshot();

    const tree2 = createRouterSnapshot(<ExpandableSource code={CODE} oneLineCode={CODE_LINE} />);
    expect(tree2).toMatchSnapshot();
  });

  it('should render correctly when not collapsed', () => {
    const tree = createRouterSnapshot(<ExpandableSource code={CODE} oneLineCode={CODE_LINE} defaultCollapsed={false} />);
    expect(tree).toMatchSnapshot();
  });
});
