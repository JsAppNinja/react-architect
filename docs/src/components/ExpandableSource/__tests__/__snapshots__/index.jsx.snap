// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandableSource should render correctly when not collapsed 1`] = `
<div>
  <div
    aria-pressed={false}
    className="md-fake-btn md-pointer--hover md-fake-btn--no-outline markdown-container md-paper md-paper--1 expandable-source"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<pre class=\\"line-numbers code-toolbar language-scss\\"><code class=\\"language-scss\\"><span class=\\"token keyword\\">@mixin</span> <span class=\\"token function\\">react-md-buttons</span><span class=\\"token punctuation\\">(</span>
  <span class=\\"token property\\"><span class=\\"token variable\\">$primary-color</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-primary-color</span>,
  <span class=\\"token property\\"><span class=\\"token variable\\">$secondary-color</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-secondary-color</span>,
  <span class=\\"token property\\"><span class=\\"token variable\\">$light-theme</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-light-theme</span>,
  <span class=\\"token property\\"><span class=\\"token variable\\">$include-media</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-media-included</span>,
  <span class=\\"token property\\"><span class=\\"token variable\\">$include-flat</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-include-flat</span>,
  <span class=\\"token property\\"><span class=\\"token variable\\">$include-raised</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-include-raised</span>,
  <span class=\\"token property\\"><span class=\\"token variable\\">$include-icon</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-include-icon</span>,
  <span class=\\"token property\\"><span class=\\"token variable\\">$include-floating</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-include-floating</span>,
  <span class=\\"token property\\"><span class=\\"token variable\\">$include-dense-icons</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-font-icon-include-dense</span>
<span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span>
  <span class=\\"token comment\\">// scss-lint:disable QualifyingElement</span>
  <span class=\\"token selector\\">a.md-btn </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token property\\">text-decoration</span><span class=\\"token punctuation\\">:</span> none<span class=\\"token punctuation\\">;</span>
  <span class=\\"token punctuation\\">}</span>

  <span class=\\"token selector\\">.md-btn </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token property\\">background</span><span class=\\"token punctuation\\">:</span> transparent<span class=\\"token punctuation\\">;</span>
    <span class=\\"token property\\">border</span><span class=\\"token punctuation\\">:</span> <span class=\\"token number\\">0</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token property\\">position</span><span class=\\"token punctuation\\">:</span> relative<span class=\\"token punctuation\\">;</span>
    <span class=\\"token property\\">transition-duration</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-transition-time</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token property\\">transition-property</span><span class=\\"token punctuation\\">:</span> background, color<span class=\\"token punctuation\\">;</span>

    <span class=\\"token comment\\">// Prevents click event from firing when clicking a child</span>
    <span class=\\"token selector\\"><span class=\\"token parent important\\">&amp;</span>[disabled] * </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token property\\">pointer-events</span><span class=\\"token punctuation\\">:</span> none<span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token selector\\"><span class=\\"token parent important\\">&amp;</span>:focus </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token property\\">outline-style</span><span class=\\"token punctuation\\">:</span> none<span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token selector\\">.md-icon </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token property\\">color</span><span class=\\"token punctuation\\">:</span> inherit<span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token selector\\">.md-icon-separator </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token property\\">height</span><span class=\\"token punctuation\\">:</span> <span class=\\"token number\\">100%</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token selector\\">.md-icon-text </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token keyword\\">@extend</span> <span class=\\"token number\\">%</span>md-font-medium<span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>
  <span class=\\"token punctuation\\">}</span>

  <span class=\\"token selector\\">.md-btn--hover </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token property\\">background</span><span class=\\"token punctuation\\">:</span> <span class=\\"token function\\">if</span><span class=\\"token punctuation\\">(</span><span class=\\"token variable\\">$light-theme</span>, <span class=\\"token variable\\">$md-btn-light-theme-hover-color</span>, <span class=\\"token variable\\">$md-btn-dark-theme-hover-color</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
  <span class=\\"token punctuation\\">}</span>

  <span class=\\"token selector\\">.md-btn--color-primary-active </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token property\\">background</span><span class=\\"token punctuation\\">:</span> <span class=\\"token function\\">rgba</span><span class=\\"token punctuation\\">(</span><span class=\\"token variable\\">$primary-color</span>, <span class=\\"token number\\">.12</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
  <span class=\\"token punctuation\\">}</span>

  <span class=\\"token selector\\">.md-btn--color-secondary-active </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token property\\">background</span><span class=\\"token punctuation\\">:</span> <span class=\\"token function\\">rgba</span><span class=\\"token punctuation\\">(</span><span class=\\"token variable\\">$secondary-color</span>, <span class=\\"token number\\">.12</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
  <span class=\\"token punctuation\\">}</span>

  <span class=\\"token keyword\\">@if</span> <span class=\\"token selector\\"><span class=\\"token variable\\">$include-flat</span> or <span class=\\"token variable\\">$include-raised</span> </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token selector\\">.md-btn--text </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token property\\">border-radius</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-text-border-radius</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">font-size</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-text-font-size</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">font-weight</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-font-medium</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">height</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-text-height</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">min-width</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-min-width</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">padding</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-tb-padding</span> <span class=\\"token variable\\">$md-btn-lr-padding</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">text-transform</span><span class=\\"token punctuation\\">:</span> uppercase<span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>
  <span class=\\"token punctuation\\">}</span>

  <span class=\\"token keyword\\">@if</span> <span class=\\"token selector\\"><span class=\\"token variable\\">$include-raised</span> or <span class=\\"token variable\\">$include-floating</span> </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token selector\\">.md-btn--raised </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token keyword\\">@include</span> <span class=\\"token function\\">md-box-shadow</span><span class=\\"token punctuation\\">(</span><span class=\\"token number\\">1</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>

      <span class=\\"token property\\">transition</span><span class=\\"token punctuation\\">:</span> background <span class=\\"token variable\\">$md-transition-time</span>, box-shadow <span class=\\"token variable\\">$md-transition-time</span> <span class=\\"token operator\\">*</span> <span class=\\"token number\\">2</span>, color <span class=\\"token variable\\">$md-transition-time</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token selector\\">.md-btn--raised-disabled </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token property\\">background</span><span class=\\"token punctuation\\">:</span> <span class=\\"token function\\">get-color</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'divider'</span>, <span class=\\"token variable\\">$light-theme</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token selector\\">.md-btn--raised-pressed </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token keyword\\">@include</span> <span class=\\"token function\\">md-box-shadow</span><span class=\\"token punctuation\\">(</span><span class=\\"token number\\">2</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>
  <span class=\\"token punctuation\\">}</span>

  <span class=\\"token keyword\\">@if</span> <span class=\\"token selector\\"><span class=\\"token variable\\">$include-icon</span> or <span class=\\"token variable\\">$include-floating</span> </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token selector\\">.md-btn--icon </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token property\\">border-radius</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-icon-border-radius</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">color</span><span class=\\"token punctuation\\">:</span> <span class=\\"token function\\">get-color</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'icon'</span>, <span class=\\"token variable\\">$light-theme</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">height</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-icon-size</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">padding</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-icon-padding</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">width</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-icon-size</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>
  <span class=\\"token punctuation\\">}</span>

  <span class=\\"token keyword\\">@if</span> <span class=\\"token selector\\"><span class=\\"token variable\\">$include-floating</span> </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token selector\\">.md-btn--floating </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token keyword\\">@include</span> <span class=\\"token function\\">md-box-shadow</span><span class=\\"token punctuation\\">(</span><span class=\\"token number\\">2</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>

      <span class=\\"token property\\">height</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-floating-size</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">padding</span><span class=\\"token punctuation\\">:</span> <span class=\\"token punctuation\\">(</span><span class=\\"token variable\\">$md-btn-floating-size</span> <span class=\\"token operator\\">-</span> <span class=\\"token variable\\">$md-font-icon-size</span><span class=\\"token punctuation\\">)</span> <span class=\\"token operator\\">/</span> <span class=\\"token number\\">2</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">transition-property</span><span class=\\"token punctuation\\">:</span> background, box-shadow, color<span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">width</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-floating-size</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token selector\\">.md-btn--floating-pressed </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token keyword\\">@include</span> <span class=\\"token function\\">md-box-shadow</span><span class=\\"token punctuation\\">(</span><span class=\\"token number\\">4</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token selector\\">.md-btn--floating-mini </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token property\\">height</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-floating-mini-size</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">padding</span><span class=\\"token punctuation\\">:</span> <span class=\\"token punctuation\\">(</span><span class=\\"token variable\\">$md-btn-floating-mini-size</span> <span class=\\"token operator\\">-</span> <span class=\\"token variable\\">$md-font-icon-size</span><span class=\\"token punctuation\\">)</span> <span class=\\"token operator\\">/</span> <span class=\\"token number\\">2</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">width</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-floating-mini-size</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token selector\\">.md-btn--fixed </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token property\\">position</span><span class=\\"token punctuation\\">:</span> fixed<span class=\\"token punctuation\\">;</span>
      <span class=\\"token property\\">z-index</span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-fixed-z-index</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>

    <span class=\\"token keyword\\">@if</span> <span class=\\"token selector\\"><span class=\\"token variable\\">$md-btn-floating-margin</span> != null </span><span class=\\"token punctuation\\">{</span>
      <span class=\\"token keyword\\">@include</span> <span class=\\"token function\\">react-md-button-fixed-positions</span><span class=\\"token punctuation\\">(</span><span class=\\"token variable\\">$md-btn-floating-margin</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
    <span class=\\"token punctuation\\">}</span>
  <span class=\\"token punctuation\\">}</span>

  <span class=\\"token keyword\\">@if</span> <span class=\\"token selector\\"><span class=\\"token variable\\">$include-media</span> </span><span class=\\"token punctuation\\">{</span>
    <span class=\\"token keyword\\">@include</span> <span class=\\"token function\\">react-md-buttons-media</span><span class=\\"token punctuation\\">(</span><span class=\\"token variable\\">$include-flat</span>, <span class=\\"token variable\\">$include-raised</span>, <span class=\\"token variable\\">$include-icon</span>, <span class=\\"token variable\\">$include-floating</span>, <span class=\\"token variable\\">$include-dense-icons</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
  <span class=\\"token punctuation\\">}</span>
<span class=\\"token punctuation\\">}</span>
<span aria-hidden=\\"true\\" class=\\"line-numbers-rows\\"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code><div class=\\"toolbar\\"><div class=\\"toolbar-item\\"><span>Sass (Scss)</span></div></pre>
",
      }
    }
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    style={undefined}
    tabIndex={0}
  />
  <h5
    className="md-cell--right md-color--secondary-text md-text-right expandable-source__label"
  >
    Click to collapse
  </h5>
</div>
`;

exports[`ExpandableSource should render correctly while collapsed 1`] = `
<div
  aria-pressed={false}
  className="md-fake-btn md-fake-btn--no-outline markdown-container md-paper md-paper--1 expandable-source"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<pre class=\\"line-numbers code-toolbar language-scss\\"><code class=\\"language-scss\\"><span class=\\"token property\\"><span class=\\"token variable\\">$md-primary-color</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-red-500</span><span class=\\"token punctuation\\">;</span>
<span aria-hidden=\\"true\\" class=\\"line-numbers-rows\\"><span></span></span></code><div class=\\"toolbar\\"><div class=\\"toolbar-item\\"><span>Sass (Scss)</span></div></pre>
",
    }
  }
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={null}
/>
`;

exports[`ExpandableSource should render correctly while collapsed 2`] = `
<div>
  <div
    aria-pressed={false}
    className="md-fake-btn md-pointer--hover md-fake-btn--no-outline markdown-container md-paper md-paper--1 expandable-source"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<pre class=\\"line-numbers code-toolbar language-scss\\"><code class=\\"language-scss\\"><span class=\\"token keyword\\">@mixin</span> <span class=\\"token function\\">react-md-buttons</span><span class=\\"token punctuation\\">(</span><span class=\\"token property\\"><span class=\\"token variable\\">$primary-color</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-primary-color</span>, <span class=\\"token property\\"><span class=\\"token variable\\">$secondary-color</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-secondary-color</span>, <span class=\\"token property\\"><span class=\\"token variable\\">$light-theme</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-light-theme</span>, <span class=\\"token property\\"><span class=\\"token variable\\">$include-media</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-media-included</span>, <span class=\\"token property\\"><span class=\\"token variable\\">$include-flat</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-include-flat</span>, <span class=\\"token property\\"><span class=\\"token variable\\">$include-raised</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-include-raised</span>, <span class=\\"token property\\"><span class=\\"token variable\\">$include-icon</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-include-icon</span>, <span class=\\"token property\\"><span class=\\"token variable\\">$include-floating</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-btn-include-floating</span>, <span class=\\"token property\\"><span class=\\"token variable\\">$include-dense-icons</span></span><span class=\\"token punctuation\\">:</span> <span class=\\"token variable\\">$md-font-icon-include-dense</span><span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span> <span class=\\"token number\\">...</span> <span class=\\"token punctuation\\">}</span>
<span aria-hidden=\\"true\\" class=\\"line-numbers-rows\\"><span></span></span></code><div class=\\"toolbar\\"><div class=\\"toolbar-item\\"><span>Sass (Scss)</span></div></pre>
",
      }
    }
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    style={undefined}
    tabIndex={0}
  />
  <h5
    className="md-cell--right md-color--secondary-text md-text-right expandable-source__label"
  >
    Click to expand
  </h5>
</div>
`;
